lib/tiny_tds/client.rb

```diff
@@ -48,8 +48,10 @@ def initialize(opts = {})
       opts[:encoding] = opts[:encoding].nil? || opts[:encoding].casecmp('utf8').zero? ? 'UTF-8' : opts[:encoding].upcase
       opts[:port] ||= 1433
       opts[:dataserver] = "#{opts[:host]}:#{opts[:port]}" if opts[:dataserver].to_s.empty?
-      forced_integer_keys = [:login_timeout, :port, :timeout]
+      forced_integer_keys = [:login_timeout, :port]
       forced_integer_keys.each { |k| opts[k] = opts[k].to_i if opts[k] }
+      forced_string_keys = [:timeout]
+      forced_string_keys.each { |k| opts[k] = opts[k].to_s if opts[k] }
```

Comment:
Given that we likely continue to want this to be integers and passed down to `dbsettime(NUM2INT(timeout))` I would suggest removing this and doing the string coercion in client.c within the `dbsetopt` call.